src/hydra/dataProvider.js

```diff
@@ -564,5 +596,61 @@ export default (
                   (status ? `Status: ${status}` : ''),
               );
             }),
+    subscribe: (resourceIds, callback) => {
+      if (!mercure || !mercure.hub) {
+        throw new Error(
+          'Mercure URL not set, did you forget to pass a `mercure` configuration to `HydraAdmin`?',
+        );
+      }
+
+      resourceIds.forEach((resourceId) => {
+        const sub = subscriptions[resourceId];
+        if (sub !== undefined) {
+          sub.count++;
+          return;
+        }
+
+        const url = new URL(mercure.hub, window.origin);
+        url.searchParams.append(
+          'topic',
+          new URL(resourceId, mercure.topicUrl).toString(),
+        );
+
+        if (mercure.jwt !== null) {
+          document.cookie = `mercureAuthorization=${mercure.jwt}; Path=${mercure.hub}; Secure; SameSite=None`;
+        }
+
+        const eventSource = new EventSource(url.toString(), {
+          withCredentials: mercure.jwt !== null,
+        });
+        const eventListener = (event) => {
+          const document = transformJsonLdDocumentToReactAdminDocument(
+            JSON.parse(event.data),
+          );
+          // the only need for this callback is for accessing redux's `dispatch` method to update RA's state.
+          callback(document);
+        };
+        eventSource.addEventListener('message', eventListener);
+
+        subscriptions[resourceId] = sub;
+      });
+
+      return Promise.resolve({ data: null });
+    },
+    unsubscribe: (resource, resourceIds) => {
+      resourceIds.forEach((resourceId) => {
+        const sub = subscriptions[resourceId];
+        if (sub !== undefined) {
+          sub.count--;
+
+          if (sub.count <= 0) {
+            sub.eventSource.removeEventListener('message', sub.eventListener);
+            delete subscriptions[resourceId];
```

Comment:
I used `delete` here by default, but I know it is slower than setting to `undefined`, so I can swap if it's preferred :slightly_smiling_face:

Reply:
I don't think we need micro-optimizations like this. It's alright like this.